internal/services/codesigning/trusted_signing_account_resource.go

```diff
@@ -0,0 +1,221 @@
+			}
+
+			meta.SetID(id)
+			return nil
+		},
+	}
+}
+
+func (m TrustedSigningAccountResource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, meta sdk.ResourceMetaData) error {
+			id, err := codesigningaccounts.ParseCodeSigningAccountID(meta.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			client := meta.Client.CodeSigning.Client.CodeSigningAccounts
+			result, err := client.Get(ctx, *id)
+			if err != nil {
+				return err
+			}
+
+			if result.Model == nil {
+				return fmt.Errorf("retrieving %s got nil model", id)
+			}
+			var output TruestedSigningAccountModel
+			output.Name = id.CodeSigningAccountName
+			output.ResourceGroupName = id.ResourceGroupName
+
+			if result.Model == nil {
+				return fmt.Errorf("Get response nil Model")
+			}
+			model := result.Model
+			output.Location = model.Location
+			output.Tags = pointer.From(model.Tags)
+			if ptrProp := model.Properties; ptrProp != nil {
+				itemProp := *ptrProp
+				output.AccountUri = pointer.From(itemProp.AccountUri)
+				if ptrSku := itemProp.Sku; ptrSku != nil {
+					itemSku := *ptrSku
+					output.SkuName = string(itemSku.Name)
+				}
+			}
+
+			return meta.Encode(&output)
+		},
+	}
+}
+
+func (m TrustedSigningAccountResource) Update() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 10 * time.Minute,
+		Func: func(ctx context.Context, meta sdk.ResourceMetaData) (err error) {
+			client := meta.Client.CodeSigning.Client.CodeSigningAccounts
+			id, err := codesigningaccounts.ParseCodeSigningAccountID(meta.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+			var model TruestedSigningAccountModel
+			if err = meta.Decode(&model); err != nil {
+				return fmt.Errorf("decoding %s: %+v", id, err)
+			}
+
+			var upd codesigningaccounts.CodeSigningAccountPatch
+			if meta.ResourceData.HasChange("sku_name") {
+				upd.Properties = pointer.To(codesigningaccounts.CodeSigningAccountPatchProperties{
+					Sku: pointer.To(codesigningaccounts.AccountSkuPatch{
+						Name: pointer.To(codesigningaccounts.SkuName(model.SkuName)),
+					}),
+				})
+			}
+			if meta.ResourceData.HasChange("tags") {
+				upd.Tags = pointer.To(model.Tags)
+			}
+
+			if err = client.UpdateThenPoll(ctx, *id, upd); err != nil {
+				return fmt.Errorf("updating %s: %v", id, err)
+			}
+
+			return nil
+		},
+	}
+}
+
+func (m TrustedSigningAccountResource) Delete() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 10 * time.Minute,
+		Func: func(ctx context.Context, meta sdk.ResourceMetaData) error {
+			id, err := codesigningaccounts.ParseCodeSigningAccountID(meta.ResourceData.Id())
+			if err != nil {
+				return err
+			}
+
+			meta.Logger.Infof("deleting %s", id)
+			client := meta.Client.CodeSigning.Client.CodeSigningAccounts
+			if _, err = client.Delete(ctx, *id); err != nil {
+				return fmt.Errorf("deleting %s: %v", id, err)
+			}
+			return nil
```

Comment:
Please consider the ordering of variable declarations within the CRUD functions, there is no good reason to reorder this when 99% of the provider follows the sequence suggested below.
```suggestion
			client := meta.Client.CodeSigning.Client.CodeSigningAccounts
			id, err := codesigningaccounts.ParseCodeSigningAccountID(meta.ResourceData.Id())
			if err != nil {
				return err
			}

			meta.Logger.Infof("deleting %s", id)
			
			if _, err = client.Delete(ctx, *id); err != nil {
				return fmt.Errorf("deleting %s: %v", id, err)
			}
			return nil
```